server/src/main/java/org/opensearch/search/backpressure/stats/SearchTaskStats.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.backpressure.stats;
+
+import org.opensearch.common.collect.MapBuilder;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.search.backpressure.trackers.CpuUsageTracker;
+import org.opensearch.search.backpressure.trackers.ElapsedTimeTracker;
+import org.opensearch.search.backpressure.trackers.HeapUsageTracker;
+import org.opensearch.search.backpressure.trackers.TaskResourceUsageTracker;
+import org.opensearch.search.backpressure.trackers.TaskResourceUsageTrackerType;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Stats related to cancelled search tasks.
+ */
+
+public class SearchTaskStats implements ToXContentObject, Writeable {
```

Comment:
This is identical to `SearchShardTaskStats`, please extract the common subclass

Reply:
Makes sense. Replaced both the classes with a common class `SearchBackpressureTaskStats`.

Reply:
@PritLadani I don't see `SearchBackpressureTaskStats` anywhere

Reply:
Well, I removed it in the subsequent commits since it might cause BWC issues in future. We may want to have different structure for SearchTaskStats and SearchShardTask stats and hence kept them separate.